meshtastic/tests/test_powermon_power_supply.py

```
@@ -159,3 +160,25 @@ def test_powermon_public_exports_remain_available() -> None:
+def test_powermon_optional_backends_are_lazy_and_dependency_error_is_clear(
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    """Optional backend access should not require dependencies at package import time."""
+    for backend_name in ("PPK2PowerSupply", "RidenPowerSupply"):
+        powermon.__dict__.pop(backend_name, None)
+
+    real_import_module = importlib.import_module
+
+    def _fake_import_module(name: str, package: str | None = None) -> object:
+        if package == "meshtastic.powermon" and name in (".ppk2", ".riden"):
+            raise ImportError("optional backend missing")
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)